src/physics/em/detail/RayleighInteractor.hh

```diff
@@ -24,49 +23,78 @@ namespace detail
 /*!
  * Brief class description.
  *
- * This is a model for XXXX process. Additional description
+ * This is a model for the Rayleigh scattering process for photons.
  *
  * \note This performs the same sampling routine as in Geant4's
- * XXXX class, as documented in section XXX of the Geant4 Physics
- * Reference (release 10.6).
+ * G4LivermoreRayleighModel class, as documented in section 6.2.2 of the
+ * Geant4 Physics Reference (release 10.6).
  */
 class RayleighInteractor
 {
+    //!@{
+    //! Type aliases
+    using ItemIdT = celeritas::ItemId<unsigned int>;
+    //!@}
+
   public:
     // Construct with shared and state data
     inline CELER_FUNCTION
-    RayleighInteractor(const RayleighInteractorPointers& shared,
-                       const ParticleTrackView&          particle,
-                       const Real3&                      inc_direction,
-                       StackAllocator<Secondary>&        allocate);
+    RayleighInteractor(const RayleighNativeRef& shared,
+                       const ParticleTrackView& particle,
+                       const Real3&             inc_direction,
+                       ElementId                element_id);
 
     // Sample an interaction with the given RNG
     template<class Engine>
     inline CELER_FUNCTION Interaction operator()(Engine& rng);
 
     //// COMMON PROPERTIES ////
 
-    //! Minimum incident energy for this model to be valid
+    //! Minimum incident energy for this model to be valid: 10 * eV
     static CELER_CONSTEXPR_FUNCTION units::MevEnergy min_incident_energy()
     {
-        return units::MevEnergy{0}; // XXX
+        return units::MevEnergy{1.0e-5};
     }
 
-    //! Maximum incident energy for this model to be valid
+    //! Maximum incident energy for this model to be valid: 1 * GeV
     static CELER_CONSTEXPR_FUNCTION units::MevEnergy max_incident_energy()
     {
-        return units::MevEnergy{0}; // XXX
+        return units::MevEnergy{1.0e+3};
+    }
+
+  private:
+    //! cm/hc in the MeV energy unit
+    static CELER_CONSTEXPR_FUNCTION real_type hc_factor()
+    {
+        return units::centimeter * unit_cast(units::MevMomentum{1.0})
```

Comment:
I think maybe this should be `unit_cast(units::MevEnergy{1.0}`?

Reply:
@amandalund You are absolutely right.  Nice catch and you save may day!

Reply:
Ahhh good catch indeed! I missed it because I was thinking of the recent mag field stuff :)

@whokion  you can also write this as
```
return units::centimeter * units::Mev::value();
```